### PR TITLE
Add remove todos

### PR DESCRIPTION
**Pull request by osamaashraf6**
## What does this PR do?

This PR is raised to make todos functionality

## Description of the task to be completed

- [x] Make a method for adding todo
- [x] Make a method for removing todo
- [x] Make a method for editing todo 

  
## How should it be tested

Clone the repository
```
git clone git@github.com:osamaashraf6/Todo-list-webpack.git
```
Navigate to the folder
```
cd repo-name
```
Checkout the branch
```
git checkout branch_name
```

#### Thanks for reviewing my PR 😊

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+//   const completed = false;
+//   const index = todosList.list.length + 1;
+//   const id = Math.floor(Math.random() * 1000);
+//   const newTodo = { description, completed, index, id };
+//   if (description) {
+//     todosList.addTodo(newTodo);
+//     render(todosList);
+//     document.querySelector(".input-todo").value = "";
+//   }
+// });
```
Comment (AmaduKamara) [inline src/index.js]: - [ ] Kindly remove all commented codes you're not using in your project. It's a good convention to have your code clean and lean. Having codes that are commented on and not meaningful to the project makes your codebase dirty. Please take them off 👍

File: src/todos.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  editTodo(todoId, todoDescription) {
+    const newData = this.list.map((todo) => {
+      if (todo.index === todoId) {
+        return { ...todo, description: todoDescription };
+      }
+      return todo;
+    });
+    localStorage.setItem('todos', JSON.stringify(newData));
+  }
```
Comment (AmaduKamara) [inline src/todos.js]: - [ ] When I try to edit a task description, it stays fine on the screen but after reloading, nothing is saved to localStorage. Kindly work on the edit functionality for the task description and get the edited values saved to LlocalStorage which is again rendered to the screen when the page reloads. 👍

image

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️♻️♻️ 

Hi @osamaashraf6,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✅ You used Gitflow correctly 👍 
✅ Linter checks passed 👍 
✅ Good commit messages 👍 
✅ Good descriptive PR 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by NduatiKagiri — APPROVED** [review]
Hi @osamaashraf6 , 

Great job making the requested changes. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
minions

## Highlights ✔️
✔️ Clearly descriptive PR
✔️ Good documentation
✔️ Linters checks are passing

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/todos.js (hunk 2 above)
Comment (osamaashraf6) [reply src/todos.js]: - Thanks 🙏 for reviewing my pull request I appreciate your review 💯 
### Thanks again for reviewing my PR  and have a good day😊

**Comment by osamaashraf6** [comment]
- Thanks 🙏 for reviewing my pull request I appreciate your review 💯 
### Thanks again for reviewing my PR  and have a good day😊

**Pull request merged**